eisforellen/thoughtlog
# Add Authentication

## What does this PR do?
This PR adds authentication using biometrics or passcode whenever the app is brought to the foreground. 

## How should this be manually tested?
Confirm that the Protected View is displayed every time the app is launched or enters the foreground.


## Any background context you want to provide?
An AuthenticationManager was added as well as a Protected View Controller. The Protected View Controller owns the presentation of the protected view, including prompting for auth. 
Also took the opportunity to clean up some of the code as well.


## Screenshots (if applicable)
![auth](https://user-images.githubusercontent.com/18013345/32901153-2f2e1b8e-cabd-11e7-88c7-cce50a1fb683.gif)
